### PR TITLE
Replace XFAIL with narrow skips to not clutter CI logs

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

Some of the XFAIL tests are now producing logs for more than a year.
Replace the XFAIL with narrow clear skips. Since xfail seems to be
mostly used to track external issues in some k8s versions, replacing
them with a skip with the implicit feature that it will break when we
introduce new k8s versions in CI which may have this feature fixed.

It is not optimal, since someone completely outside of the project may
want to run tests against their cluster and they may be confused, but it
adds a repeating non-ignorable re-evaluation point while keeping clear
what's supposed to work and what not.

For the seldom other cases where xfail was used I would recommend a clear skip pointing to an issue which is assigned to someone, since it does not seem to work out to get rid of xfails fast, which is not helpful when changing code or when refactoring.

Getting rid of xfail resolved the following issues:
 * whenever something really fails in that area it is very hard to understand if xfail tests are now failing more
 * it was still confusing for people  to read (X)FAIL during  their code changes, making it not clear to them if they should ignore it or not.
 * XFAIL does not seem to speed up fixing issues, which means that all engineers are exposed to additional CI logs which they have to parse for quite some time, increasing the cognitive load.
 * Since the skips were just continuing to happen on newer k8s versions it was not clear if failing tests still fail due to the original issue, but since CI jobs stayed green the incentive to check the tests again was low. As a result it was not clear what was supposed to work and what not.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

@AlonaKaplan it turns out that starting with k8s-1.22 most tests finally start to succeed. This finding is considered in the skips.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
> I agree that since the tests in the `expose` file are working since k8s 1.22, it makes sense to substitute them with skips. But why removing the `XFail` implementation? I suggest splitting the PR. Keeping the `expose` tests changes. And discussing the `XFail` complete removal in another PR. @EdDev

I was sceptical in letting it being added and I think it clearly did not add much benefit. Instead I got asked multiple times if the XFail is something to bother. During the removal I just found out that some of the XFails don't fail always anymore (which is again a hint that they don't add the value which @EdDev had in mind).

File: tests/network/expose.go [hunk 1]
```diff
@@ -187,6 +196,7 @@ var _ = SIGDescribe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:c
 			})
 
 			DescribeTable("[label:masquerade_binding_connectivity]Should expose a Cluster IP service on a VMI and connect to it", func(ipFamily ipFamily) {
+				skipIfDualStackAndVersionBelow(ipFamily, skipBrokenDualStackChecks, k8s122)
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/network/expose.go:199]: Yes, it just seems to me that these two test cases: 

```
  Entry("[test_id:1533] over default IPv4 IP family", ipv4),
  Entry(overIPv6Family, ipv6),
```

test exactly the same scenario like these two test cases with the XFail:

```
  Entry(overDualStackIPv4, dualIPv4Primary),
  Entry(overDualStackIPv6, dualIPv6Primary),
```

on a dual-stack cluster which means that there is not really an additional value in these cases if the cluster does not propery support dual-stack services? But maybe I misunderstand something. In this case I would of course change it.

[the remark this answers is not in the document]
Comment (AlonaKaplan) [reply tests/network/expose.go:199]: Yes, it makes sense what you're saying. There is no good reason to use a dual stack service if it is half broken...
But maybe we have users that don't even know their service is dual stack and are using just the primary network.
It is working, so why not test it?

[the remark this answers is not in the document]
Comment (rmohr) [reply tests/network/expose.go:199]: Makes sense. Will change it. I now understand that the service is created differently (different kubectl commandline).

File: tests/network/expose.go [hunk 2]
```diff
@@ -68,6 +67,12 @@ func isDualStack(ipFamily ipFamily) bool {
 	return ipFamily == dualIPv4Primary || ipFamily == dualIPv6Primary
 }
 
+func skipIfDualStackAndVersionBelow(stack ipFamily, reason string, maxK8sVersion string) {
+	if isDualStack(stack) {
+		checks.SkipIfVersionBelow(reason, maxK8sVersion)
+	}
+}
+
```
Comment (fossedihelm) [inline tests/network/expose.go:75]: How about moving this function inside `tests/framework/checks/skips.go`? Thanks

**Pull request closed without merging**